sbom/examples/container_image/release/from_catalog.py

```diff
@@ -181,6 +189,43 @@ def generate_sboms_for_image(image_nvr):
             image_pkg["externalRefs"].append(ref)
         per_arch_images.append(image_pkg)
 
+        # Add in parent images
+        parent_images = koji_session.getBuild(image_nvr)
+        for key in ("extra", "typeinfo", "image", "parent_images"):
+            parent_images = parent_images.get(key, {})
+
+        parent_images = [img.rsplit("/")[-1] for img in parent_images if img != "scratch"]
+        direct_parent_index = len(parent_images) - 1
+        for index, parent_image in enumerate(parent_images):
+            parent_spdx_id = f"SPDXRef-parent-image-{index}-{image['architecture']}"
+            parent_pkg = {
+                "SPDXID": parent_spdx_id,
+                "name": f"{parent_image}_{image['architecture']}",
+                "versionInfo": "NOASSERTION",
+                "supplier": "Organization: Red Hat",
+                "downloadLocation": "NOASSERTION",
+                "licenseDeclared": "NOASSERTION",
+                "externalRefs": [],
+            }
+            other_pkgs.append(parent_pkg)
+
+            if index == direct_parent_index:
+                other_rels.append(
+                    {
+                        "spdxElementId": spdx_image_id,
+                        "relationshipType": "DESCENDANT_OF",
+                        "relatedSpdxElement": parent_spdx_id,
+                    }
+                )
+            else:
+                other_rels.append(
+                    {
+                        "spdxElementId": parent_spdx_id,
+                        "relationshipType": "BUILD_TOOL_OF",
+                        "relatedSpdxElement": spdx_image_id,
```

Comment:
If I take e.g.:

```
rhel-els
└── openshift-base-rhel9
    └── openshift-enterprise-base-rhel9
        └── ose-etcd
```

and we're manifesting `ose-etcd`, would `openshift-enterprise-base-rhel9` and `openshift-base-rhel9` be the "build tool of" images and `ose-etcd` be the descendant of `rhel-els`?

Reply:
No, they are all descendants. The BUILD_TOOL_OF relationship here describes multistage builds.

If a Dockerfile says:
```
FROM img-a AS stage1
RUN make foo

FROM img-b AS stage2
RUN make bar

FROM ubi9
COPY --from=stage1 /work/foo /foo
COPY --from=stage2 /work/bar /bar
```

then img-a and img-b would be described as build tooling for the final image, whereas it would be a descendant of ubi9.

Reply:
Gotcha! That makes sense!